dev/src/glloadiqm.cpp

```diff
@@ -357,3 +359,119 @@ Mesh *LoadIQM(string_view filename) {
+    vector<float3> vpositions;
+    vector<float3> vnormals;
+    vector<byte4> vcolors;
+    vector<iqmtriangle> triangles;
+
+    // Prepare vertex data
+    vattribs.resize(geom->nverts);
+    vpositions.reserve(geom->nverts);
+    vnormals.reserve(geom->nverts);
+    vcolors.reserve(geom->nverts);
+    GL_CALL(glBindBuffer(GL_ARRAY_BUFFER, geom->vbo1));
+    GL_CALL(glGetBufferSubData(GL_ARRAY_BUFFER, 0, geom->nverts*sizeof(vert), vattribs.data()));
+    for (const auto &v : vattribs) {
+        vpositions.push_back(float3(v.pos[0], v.pos[1], v.pos[2]));
+        vnormals.push_back(float3(v.normal[0], v.normal[1], v.color[2]));
+        vcolors.push_back(byte4(v.color[0], v.color[1], v.color[2], v.color[3]));
+    }
+
+    // Prepare triangle data
+    memset(&mesh2, 0, sizeof(mesh2));
+    mesh2.num_vertexes = geom->nverts;
+    mesh2.num_triangles = surf->numidx / 3;
+    triangles.resize(mesh2.num_triangles);
+    GL_CALL(glBindBuffer(GL_ARRAY_BUFFER, surf->ibo));
+    GL_CALL(glGetBufferSubData(GL_ARRAY_BUFFER, 0, mesh2.num_triangles*sizeof(iqmtriangle), triangles.data()));
+
+    // Init header
+    memset(&hdr, 0, sizeof(hdr));
+    memcpy(hdr.magic, IQM_MAGIC, sizeof(hdr.magic)); 
+    hdr.filesize = sizeof(hdr);
+    hdr.version = IQM_VERSION;
+    hdr.num_meshes = 1;
+    hdr.ofs_meshes = hdr.filesize;
+    hdr.filesize += sizeof(iqmmesh);
+    uint32_t voffset = hdr.filesize + sizeof(varray);
+    hdr.num_vertexarrays = 3;
+    hdr.ofs_vertexarrays = hdr.filesize;
+    hdr.filesize += sizeof(varray);
+    uint32_t valign = (8 - (hdr.filesize%8))%8;
+    voffset += valign;
+    hdr.filesize += valign + vattribs.size()*sizeof(vert);
+    hdr.num_vertexes = vattribs.size();
+    hdr.num_triangles = triangles.size();
+    hdr.ofs_triangles = hdr.filesize;
+    hdr.filesize += triangles.size() * sizeof(iqmtriangle);
+
+    // Init vertexarray(s)
+    memset(varray, 0, sizeof(varray));
+    varray[0].type = IQM_POSITION;
+    varray[0].format = IQM_FLOAT;
+    varray[0].size = 3;
+    varray[0].offset = voffset;
+    varray[1].type = IQM_NORMAL;
+    varray[1].format = IQM_FLOAT;
+    varray[1].size = 3;
+    varray[1].offset = voffset + vpositions.size()*sizeof(float3);
+    varray[2].type = IQM_COLOR;
+    varray[2].format = IQM_UBYTE;
+    varray[2].size = 4;
+    varray[2].offset = voffset + (vpositions.size()+vnormals.size())*sizeof(float3);
+
+    // Change to little-endian
+    if (!islittleendian()) {
+        endianswap(&hdr.version, (sizeof(hdr) - sizeof(hdr.magic))/sizeof(uint32_t));
+        endianswap((uint32_t*)&mesh2, sizeof(mesh2)/sizeof(uint32_t));
+        endianswap((uint32_t*)&varray[0], sizeof(varray)/(sizeof(uint32_t)));
+        endianswap((float*)vpositions.data(), vpositions.size()*3);
+        endianswap((float*)vnormals.data(), vnormals.size()*3);
+        endianswap((uint32_t*)triangles.data(), triangles.size()*3);
+    }
+
+    // Accumulate data to be written
+    ostringstream ss;
```

Comment:
`ostringstream` is a bit high level and entails multiple copies, easier to just use `string::append`